### ISSUE TITLE
Allow the cloud instance to create project and submit PR

### ISSUE THREAD

**Issue by pascalgrimaud** [jhipster/jhipster-lite#10599]
It's a global ticket here, which must be splitted into smaller ones

The idea is:
![image](https://github.com/user-attachments/assets/f597fca9-67a9-41a5-9c17-df0830426c09)

- The user must be logged into GitHub
- he/she can:
  - create a new one
  - select an existing projects or in his organization
  - JHLite will clone the project and display the existing modules already applied
- when apply new, JHLite will
  - create a new branch
  - apply the module(s)
  - submit a pull request

It's similar to what JHipster Online provides, we should keep the idea, but not the code

**Comment by renanfranca** [comment]
Hello,

I'm currently working on this issue and decided to start with a small feature on the backend:
- Enable generating a token using GitHub login and return that token.

As I was developing, I noticed that the `spring-boot-oauth2` module in jhipster-lite already has a pretty robust implementation for OAuth2 authentication. 
However, the goal with GitHub here is just to fetch repository information and make changes on GitHub. The aim isn't to enforce a security layer across the entire jhipster-lite app, where access to certain endpoints depends on whether you're logged in with GitHub or not.

So, I'm moving forward without using the `spring-boot-oauth2` module and instead creating a `github` module in `src/main/java/tech/jhipster/lite/shared/github`.

What do you all think? Does my approach make sense, or should I reconsider using the `spring-boot-oauth2` module?
![image](https://github.com/user-attachments/assets/ac323ab2-025b-4dc2-b2cd-37eec1d3729c)

**Comment by renanfranca** [comment]
> This ticket is a really big feature. I'm not sure we should implement it now, we should discuss about it first and estimate the real value to have it. 

@pascalgrimaud : I agree 100% that it is indeed a big feature. I haven't created any pull requests because I took a step back and I am deeply studying OAuth2 while developing a prototype to demonstrate the possible workflow I am considering implementing. This way, we can talk about whether it's worth pursuing or not.

> So for now we should focus on other small features...

This discussion [Code coverage 100% for front-end generated apps](https://github.com/jhipster/jhipster-lite/discussions/10927) is pending resolution and it is blocking other issues, for example: 
- https://github.com/jhipster/jhipster-lite/issues/10874

I can work both in backend and frontend now at jhipster-lite project. Can you suggest which issue do you like me to work on?

**Comment by renanfranca** [comment]
> This ticket is a really big feature.  
I'm not sure we should implement it now; we should discuss it first and estimate the real value of having it.  
For now, we should focus on other smaller features.  

I tested the current JHipster Online feature, and it only provides the creation of new repositories, not cloning existing ones or applying changes. Even though, when I inspected it with devtools, I found the endpoint that retrieves the list of repositories, but it is not used in the interface.  

I am finishing the prototype, and this feature will impact a lot of existing functionalities. As a consequence, it will result in a significant change to the codebase and lead to increased code complexity and maintenance costs.  

I loved the challenge of this issue. I don’t use JHipster Lite daily for coding. This feature, I think, is useful for showoff moments like presentations and so on, but for daily work, I believe I would prefer using JHipster Lite running locally.

I really want to know if this feature is truly useful and if it’s worth implementing. Please let me know what you (@pascalgrimaud , @murdos ,  @DamnClin , @qmonmert  ) think about it.

**Comment by pascalgrimaud** [comment]
It was a simple idea. It is NOT a must have.
Personaly, I won't use it, I prefer to clone and start jhlite locally

**Comment by murdos** [comment]
I personally won't use it either.
I would rather use locally a cli version of jhlite.

**Comment by pascalgrimaud** [comment]
So let's close this.
Too much effort for something which will be used only in demo / meetup / talk.

Personaly, I won't use this feature for real customer, I'd prefer to launch a local version of JHLite
